packages/internal/src/utils.js

```diff
@@ -411,37 +411,50 @@ export const assertAllDefined = obj => {
   }
 };
 
-const neverDone = harden({ done: false, value: null });
+/** @type {IteratorResult<undefined, never>} */
+const neverDone = harden({ done: false, value: undefined });
+
+/** @type {IteratorResult<never, void>} */
+const alwaysDone = harden({ done: true, value: undefined });
 
-/** @type {AsyncIterable<null>} */
 export const forever = asyncGenerate(() => neverDone);
 
 /**
- * @param {() => unknown} boolFunc
- * `boolFunc`'s return value is used for its truthiness vs falsiness.
+ * @template T
+ * @param {() => T} produce
+ * `produce`'s return value is used for its truthiness vs falsiness.
```

Comment:
The comment is no longer accurate, since it is not the truthiness of the result of `produce()`, which is a `T`, that is being tested. Rather, it is the truthiness of `await produce()`, which is an `Awaited<T>`, which is being tested.